src/modules/sdlog2/sdlog2.c

```diff
@@ -581,10 +594,19 @@ static void *logwriter_thread(void *arg)
 			should_wait = true;
 		}
 
-		if (++poll_count == 10) {
+		/* slow down fsync */
+		if (poll_count % 10 == 0) {
 			fsync(log_fd);
-			poll_count = 0;
+
+		/* slow down the check even more, and don't do both at the same time */
```

Comment:
Can we do this based on the last written bytes? Something like:

```
if (bytes_written - last_check_bytes > 20M) {
  check();
  last_check_bytes = bytes_written;
}
```
This avoids coupling the frequency of the check to the buffer size used for writing, and is closer to the intent.